src/ocm/components/clusterConfiguration/staticIp/commonValidationSchemas.tsx

```diff
@@ -144,3 +145,22 @@ export const getIpAddressInSubnetValidationSchema = (
     },
   );
 };
+
+export const getIpIsNotNetworkOrBroadcastAddressSchema = (
+  protocolVersion: 'ipv4' | 'ipv6',
+  subnet: string,
+) => {
+  return Yup.string().test(
+    'is-not-network-or-broadcast',
+    `The IP address must not match the network or broadcast address`,
+    (value) => {
+      const subnetAddr = getAddressObject(subnet, protocolVersion);
+      if (subnetAddr) {
+        const subnetStart = subnetAddr?.startAddress().correctForm();
+        const subnetEnd = subnetAddr?.endAddress().correctForm();
+        return !(value === subnetStart || value === subnetEnd);
+      }
+      return true;
```

Comment:
If `getAddressObject` [returns null](https://github.com/ammont82/assisted-ui-lib/blob/5924fe5b4c1c9ea35a58c6a0ef94ce5b90452c8a/src/ocm/components/clusterConfiguration/staticIp/data/protocolVersion.ts#L33) then [the validation passes](https://github.com/jquense/yup#schematestname-string-message-string--function--any-test-function-schema:~:text=a%20validation%20function%20that%20must%20return%20true%20when%20the%20current%20value%20is%20valid), is that what we want?

Reply:
True. New PR: https://github.com/openshift-assisted/assisted-ui-lib/pull/1601